image/image.go

```diff
@@ -160,10 +172,13 @@ func sanitize(s string) string {
 	return strings.Replace(s, "/", "-", -1)
 }
 
+// compile-time check that manifestSchema2 implements genericManifest
+var _ genericManifest = (*manifestSchema2)(nil)
+
 type manifestSchema2 struct {
-	src    types.ImageSource
-	Config descriptor   `json:"config"`
-	Layers []descriptor `json:"layers"`
+	src               types.ImageSource
+	ConfigDescriptor  descriptor   `json:"config"`
+	LayersDescriptors []descriptor `json:"layers"`
```

Comment:
Oops, I somehow thought that this is renaming the types, not the members.  Stupid me, sorry, this is fine.